pages/api/applications/index.tsx

```diff
@@ -206,6 +206,73 @@ async function handlePostApplications(req: NextApiRequest, res: NextApiResponse)
   });
 }
 
+/**
+ * Handles PUT requests to /api/applications.
+ *
+ * This updates an existing application in the database using the information
+ * provided in the request. If a user is not signed in, this route will return a
+ * 403 Unauthenticated error.
+ *
+ * @param req The HTTP request
+ * @param res The HTTP response
+ */
+async function handlePutApplications(req: NextApiRequest, res: NextApiResponse) {
+  const registrationAllowed = await checkRegistrationAllowed();
+  if (!registrationAllowed) {
+    return res.status(403).json({
+      msg: 'Registration updates are no longer allowed',
+    });
+  }
+
+  const applicationBody = req.body;
+
+  let body: Registration;
+  try {
+    body = JSON.parse(req.body);
+  } catch (error) {
+    console.error('Could not parse request JSON body');
+    return res.status(400).json({
+      type: 'invalid',
+      message: 'Invalid JSON body',
+    });
+  }
+
+  const snapshot = await db
+    .collection(APPLICATIONS_COLLECTION)
+    .where('user.id', '==', body.user.id)
+    .get();
+
+  if (snapshot.empty) {
+    return res.status(404).json({
+      msg: 'Profile does not exist',
+    });
+  }
+
+  const completedRegistrationInfo = {
+    ...body,
+    user: {
+      ...body.user,
+      permissions: ['hacker'],
+    },
+  };
+
+  await db.collection(APPLICATIONS_COLLECTION).doc(body.user.id).update(completedRegistrationInfo);
+
+  const { eligible, teamMembers } = await checkAutoAcceptEligibility(body);
```

Comment:
Might need to handle case when user lost eligibility because they changed teammate or smth.

Reply:
so do we allow them to update if it might led them to lose eligibility?

Reply:
any idea on how to handle this?

Reply:
Probably we can compare the old teammate list against the new teammate list. If the 2 lists are the same, then we do nothing. Otherwise, we run eligibility check on emails in both old and new list.  If an email is no longer eligible, we can just delete the corresponding document in `acceptreject`. 

Another way, which is way easier, is to just disallow changing email and teammate email. That way, we won't have to deal with the complexity of maintaining the invariant of team relationship.

Reply:
i disable user to change teammate email for now, need them to email hello@hackutd.co to request teammate change. maybe we can have a google form for them to submit a teammate request change, it's easier to keep track of.